specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1197 @@
+
+AppNotificationContent and Button return arguments to the activated application when the user clicks
+on the component. Developers can add multiple key/value pairs and Windows App SDK will serialize
+them into a string.
+
+Below is an example with AppNotificationContent and Button:
+
+```c#
+AppNotificationContent(
+    .AddArgument(L"action", L"appNotificationBodyTouch")
+    .AddArgument(L"user", 938163)
+    .AddButton(Button(L"Send")
+        .AddArgument(L"action", L"buttonTouch")
+        .AddArgument(L"user", 938163)));
+```
+
+When the user clicks on the AppNotification body, the resulting argument string is:
+
+-   "action=appNotificationBodyTouch;user=938163"
+
+When the user clicks on the Button, the result argument string is:
+
+-   "action=buttonTouch;callId=938163"
+
+Notice that ';' is used as the delimiter between each argument.
+
+To retrieve the arguments from the serialized string, developers will use
+AppNotificationReceivedEventArgs::Arguments to get the values from the argument string as a map.
+
+Below is an example usage:
+
+```c#
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+
+// Retrieve the activation args using Windows App SDK AppLifecycle functions
+AppActivationArguments args { AppLifecycle::AppInstance::GetCurrent().GetActivatedEventArgs() };
+ExtendedActivationKind kind { args.Kind() };
+
+// Check the activation kind
+if (kind == ExtendedActivationKind::AppNotification)
+{
+    AppNotificationActivatedEventArgs appActivatedArgs { args.Data().as<AppNotificationActivatedEventArgs>() };
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { appActivatedArgs.Argument() };
+}
+```
+
+# Full API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass AppNotificationTextProperties
+    {
+        // Contains the set of <text> attributes
+        AppNotificationTextProperties();
+
+        AppNotificationTextProperties SetLanguage(String language);
+        AppNotificationTextProperties UsesCallScenarioAlign();
+        AppNotificationTextProperties SetMaxLines(Int32 maxLines);
+
+        String GetXml();
+    }
+
+    enum ButtonStyle
+    {
+        Success, // The button will be green
+        Critical, // The button will be red
+    };
+
+    runtimeclass AppNotificationButton
+    {
+        // Button can use ToolTip instead of content which requires empty content.
+        // Argument attribute is required and uses the ArgumentSerializer construct
+        // to add arguments to the button.
+        AppNotificationButton();
+        AppNotificationButton(String content);
+
+        AppNotificationButton AddArgument(String key, String value);
+
+        // Sets the IconUri for the button.
+        AppNotificationButton SetIconUri(Windows.Foundation.Uri iconUri);
```

Comment:
consider `SetIcon`, the uri is clear given the param type, consider `value` for the param to avoid repeating terms.